src/crypto/crypto_timing.cc

```diff
@@ -17,6 +17,51 @@ using v8::Value;
 namespace crypto {
 namespace Timing {
 
+struct BigIntInfo {
+  int sign_bit;
+  int word_count;
+  std::unique_ptr<uint64_t> words;
+};
+
+void TimingSafeEqualBigInt(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  if (!args[0]->IsBigInt()) {
+    THROW_ERR_INVALID_ARG_TYPE(
+      env, "The \"val1\" argument must be a bigint.");
+    return;
+  }
+  if (!args[1]->IsBigInt()) {
+    THROW_ERR_INVALID_ARG_TYPE(
+      env, "The \"val2\" argument must be a bigint.");
+    return;
+  }
+
+  BigIntInfo inf1 {};
+  BigIntInfo inf2 {};
+
+  Local<v8::BigInt> b1 = args[0].As<v8::BigInt>();
+  Local<v8::BigInt> b2 = args[1].As<v8::BigInt>();
+
+  inf1.word_count = b1->WordCount();
+  inf2.word_count = b2->WordCount();
+
+  if (inf1.word_count != inf2.word_count)
+    return args.GetReturnValue().Set(false);
+
+  inf1.words.reset(Malloc<uint64_t>(inf1.word_count));
+  inf2.words.reset(Malloc<uint64_t>(inf1.word_count));
+
+  b1->ToWordsArray(&inf1.sign_bit, &inf1.word_count, inf1.words.get());
+  b2->ToWordsArray(&inf2.sign_bit, &inf2.word_count, inf2.words.get());
```

Comment:
I’m not sure if V8 guarantees that two equal bigint instances have the same word count (i.e. that bigints do not contain zero words at their beginning)

Reply:
We should be fine.

When creating a BitInt instance from JavaScript, unless I've missed something entirely, there's no way of creating one with extraneous zero words, and the way that the internal representation works, leading zero words are part of the value and trailing zero words are truncated.

For instance, if I create a BigInt using:

```cpp
uint64_t words[] = { 0UL, 0UL, 0UL, 1UL };
Local<BigInt> test;
if (BigInt::NewFromWords(env->context(), 0, arraysize(words), words).ToLocal(&test_bn)) {
  // ...
}
```

The value of the BigInt is 6277101735386680763835789423207666416102355444464034512896, and the word count is 4, as expected.

However, if I create a BigInt using...

```cpp
uint64_t words[] = { 1UL, 0UL, 0UL, 0UL };
Local<BigInt> test;
if (BigInt::NewFromWords(env->context(), 0, arraysize(words), words).ToLocal(&test_bn)) {
  // ...
}
```

The value of the BigInt is 1, and the word count is 1.

Reply:
I’m more concerned about cases like `let a = (1n << 64n), b = ((1n << 64n) - 1n); a -= b;` – the first value is 3 words, the second 2, and the result is just a single word, but V8 might decide that it’s not worth shrinking the memory allocated for `a` here and instead keep two leading 0 words because memory resizing, even when shrinking, is expensive.

So I’m not necessarily concerned about the current state, but also about what optimizations V8 might enable in the future.

Reply:
Let's see if we can get clarification from @nodejs/v8, however, I still think we're fine. Regardless of what V8 chooses to do internally, `BigInt::WordCount()` should still return the minimal number of words necessary to encode the value and a quick test shows that it does.

Reply:
> Let's see if we can get clarification from @nodejs/v8, however, I still think we're fine. 

Fwiw, it’s lines like these that have me worried: https://github.com/nodejs/node/blob/ed8af4e93831d3cf21d5562e900371d796b5fa20/deps/v8/src/objects/bigint.cc#L390

> Regardless of what V8 chooses to do internally, `BigInt::WordCount()` should still return the minimal number of words necessary to encode the value and a quick test shows that it does.

It just passes along whatever is stored in the internal representation – I know that because I wrote it. ;)

Reply:
> It just passes along whatever is stored in the internal representation – I know that because I wrote it. ;)

Heh... yeah I figured you may have. I was speaking more about the contract of the call and not necessarily how it's implemented internally, but we definitely should get clarity on this. Btw, this kind of thing is exactly why this is in a separate commit ;-)

Reply:
Tell you what, I don't want this PR to be held up on this particular point so I'm going to remove this commit and push it to a separate PR after we get the rest of the changes here resolved